dylint/src/lib.rs

```diff
@@ -516,8 +529,29 @@ fn check_or_fix(opts: &Dylint, resolved: &ToolchainMap) -> Result<()> {
                 (env::RUSTC_WORKSPACE_WRAPPER, &*driver.to_string_lossy()),
                 (env::RUSTUP_TOOLCHAIN, toolchain),
             ])
-            .args(args)
-            .success();
+            .args(args);
+
+        if let Some(stderr_path) = &opts.pipe_stderr {
+            let path = Path::new(&stderr_path);
+            let file = OpenOptions::new()
+                .append(true)
+                .create(true)
+                .open(path)
```

Comment:
```suggestion
                .open(&stderr_path)
```
And then remove the declaration of `path` just above. (I think this should work.)